src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -289,15 +298,15 @@ private async Task<Recommendation> GetSelectedRecommendationFromPreviousDeployme
                 throw new InvalidUserDeploymentSettingsException(DeployToolErrorCode.StackCreatedFromDifferentDeploymentRecommendation, errorMessage.Trim());
             }
 
-            // currently the ability to load and apply previous settings is available only if the deployment target is a CloudFormation Stack
-            // Do not have an easy way to capture previous settings without querying multiple APIs for non CloudFormation based deployments
-            if (deployedApplication.ResourceType != CloudApplicationResourceType.CloudFormationStack)
-                return selectedRecommendation;
+            IDictionary<string, object> previousSettings;
+            if (deployedApplication.ResourceType == CloudApplicationResourceType.CloudFormationStack)
+                previousSettings = (await _templateMetadataReader.LoadCloudApplicationMetadata(deployedApplication.Name)).Settings;
+            else
+                previousSettings = (await _deployedApplicationQueryer.GetPreviousSettings(deployedApplication));

```

Comment:
`_deployedApplicationQueryer.GetPreviousSettings` is essentially the generic function to get previous settings. Can you move `previousSettings = (await _templateMetadataReader.LoadCloudApplicationMetadata(deployedApplication.Name)).Settings;` also into that method? I don't see why they need to be separated at this point.